addons/guide/modifiers/guide_modifier_curve.gd

```diff
@@ -24,27 +27,26 @@ static func default_curve() -> Curve:
 
 
 # Initialize this resource with default curves
-func _init(_curve_x:Curve = null, _curve_y:Curve = null, _curve_z:Curve = null):
-	if _curve_x == null:
-		_curve_x = default_curve()
-
-	if _curve_y == null:
-		_curve_y = default_curve()
+func _init(_curve:Curve = null, _x:bool = true, _y:bool = false, _z:bool = false):
+	if _curve == null:
+		_curve = default_curve()
 
-	if _curve_z == null:
-		_curve_z = default_curve()
-
-	curve_x = _curve_x
-	curve_y = _curve_y
-	curve_z = _curve_z
+	curve = _curve
 
 
 func _modify_input(input:Vector3, delta:float, value_type:GUIDEAction.GUIDEActionValueType) -> Vector3:
-	var x:float = curve_x.sample(input.x) if curve_x != null else 0.0
-	var y:float = curve_y.sample(input.y) if curve_y != null else 0.0
-	var z:float = curve_z.sample(input.z) if curve_z != null else 0.0
-
-	return Vector3(x, y, z)
+	# Curve should never be null
+	assert(curve != null, "No curve added to Curve modifier")
```

Comment:
Added an assert here to ensure the curve is never null.  Saw others using `push_error` instead though...felt it was more appropriate to halt things if no curve was defined in this case as you'd just be wasting cycles for no reason.

Honestly, unsure this will ever happen due to how the initializer works, and testing it made me aware of some UI sync strangeness...

The modifier's `_init` creates a default curve for you so it's easy to modify--saves people a couple of clicks in the inspector having to create the curve and add points to it or selecting a predefined one.  However, nothing stops someone from clearing the curve resource afterwards.

If the curve is cleared, the `_init` method still instantiates a default curve at runtime, but the GUIDE UI is never aware of this.  The modifier acts correctly using the default curve even though the UI shows there is no curve defined.  I tried doing an `emit_changes()` in the `_init`, but, duh, nope, that's not going to cause the UI to update.

Is there a better callback representing "this resource has been added for the first time, and first time only"?...something that can be used to initialize the modifier once it's been added from the UI, but not every time the resource is initialized?

Will making `curve` a property with getters/setters make it easier to trigger a UI update?

Any way to mark a field as mandatory so it can't be cleared in the inspector?

How much effort is "just enough" to help prevent people from shooting themselves in the foot...and not being over engineered?